lib/core/jdl_application_factory.js

```diff
@@ -38,23 +38,20 @@ module.exports = {
  */
 function createJDLApplication(config = {}) {
   const baseConfig = getDefaultConfigForNewApplication(config);
+  let appConfig;
   switch (config.applicationType) {
     case ApplicationTypes.MICROSERVICE:
-      return new JDLApplication({
-        config: getConfigForMicroserviceApplication(baseConfig)
-      });
+      appConfig = getConfigForMicroserviceApplication(baseConfig);
+      break;
     case ApplicationTypes.GATEWAY:
-      return new JDLApplication({
-        config: getConfigForGatewayApplication(baseConfig)
-      });
+      appConfig = getConfigForGatewayApplication(baseConfig);
+      break;
     case ApplicationTypes.UAA:
-      return new JDLApplication({
-        config: getConfigForUAAApplication(baseConfig)
-      });
+      appConfig = getConfigForUAAApplication(baseConfig);
+      break;
     case ApplicationTypes.MONOLITH:
     default:
-      return new JDLApplication({
-        config: getConfigForMonolithApplication(baseConfig)
-      });
+      appConfig = getConfigForMonolithApplication(baseConfig);
   }
+  return new JDLApplication({ config: appConfig });
```

Comment:
Might as well directly return, unless this is actually a part of the other PR

Reply:
When I changed to post validation, it's not necessary for now.
Kept because can be needed to pass options to lexer.

Reply:
I am revert if you prefer.